### DISCUSSION TITLE
Why does EvoMaster in Black-Box Mode focus on 500, 401, 404, and 301 status codes and not others?

### DISCUSSION THREAD

**Discussion by VeeraPratap-AutomationQA** [EMResearch/EvoMaster#1136] [Q&A]
Dear Andrea,

I'm using EvoMaster in Black-Box (BB) mode to generate test scripts for my API. I noticed that most of the generated tests focus on status codes like 500 (Internal Server Error), 401 (Unauthorized), 404 (Not Found), and occasionally 301 (Moved Permanently). However, there are no tests for status codes in the ranges 100-199 or 200-299, such as 200 (OK) and all others.

Could you kindly clarify why this behavior occurs? Is there a specific configuration I need to adjust to include tests for informational and successful response codes?

command java -jar evomaster.jar --blackBox true --bbSwaggerUrl https://ig.aidtaas.com/pi-bigqry-service-dbas/v3/api-docs --outputFormat JAVA_JUNIT_4 --maxTime 60s --ratePerMinute 60

Thank you for your guidance and support!

Best regards,
Pratap

**Comment by arcuri82** [comment]
hi,
 there is no specific setting for it that needs to be activated. There could be few reasons to explain not getting any 2xx:
1) you running EM for too little time, with low rate per minute. with your settings, you are only making 60 calls in total. if you own that API, you can remove the --ratePerMinute, and use a longer time budget, eg,  --maxTime 10m
2) there are complex constraints that we do not manage to solve by chance, ie, with random sampling
3) constraints depend on data into databases. you can extend the OpenAPI schema with "examples" entries with valid data (but currently we have no documentation to explain this)
4) if you get 401, you have not setup authentication info for the API (see documentation regarding auth)
